src/components/user/DisplayUserSeniors.tsx

```diff
@@ -0,0 +1,78 @@
+"use client";
+
+import { editSeniorIDs } from "@api/user/[uid]/edit-seniors/route.client";
+import { UserTile } from "@components/TileGrid";
+import Assignment from "@components/senior/assignment";
+import { RoleToUrlSegment } from "@constants/RoleAlias";
+import { Prisma, Senior } from "@prisma/client";
+import { compareSenior, fullName, seniorFullName } from "@utils";
+import React from "react";
+import { UserContext } from "src/context/UserProvider";
+
+interface DisplayProps {
+  editable: boolean;
+  currentUser: Prisma.UserGetPayload<{
+    include: { Chapter: { include: { seniors: true } } };
+  }>;
+}
+
+const DisplayUserSenior = (props: DisplayProps) => {
+  const { user } = React.useContext(UserContext);
+  const { editable, currentUser } = props;
+
+  const seniors = React.useMemo(
+    () => currentUser.Chapter?.seniors.sort(compareSenior) ?? [],
+    [currentUser.Chapter?.seniors]
+  );
+
+  const getAssignments = () =>
+    seniors?.filter((senior) => currentUser.SeniorIDs.includes(senior.id));
+
+  const [assigned, setAssigned] = React.useState(() => getAssignments());
+
+  const onSave = async () => {
+    await editSeniorIDs(
+      {
+        body: {
+          SeniorIDs: assigned.map((senior) => senior.id),
+        },
+      },
+      currentUser.id
+    );
+  };
+
+  return (
+    <div className="flex flex-col gap-y-6">
+      <h1 className="text-4xl font-bold text-[#000022]">
+        {fullName(currentUser)}
+      </h1>
+      <Assignment
+        editable={editable}
+        display={(senior: Senior) => seniorFullName(senior)}
+        elements={seniors}
+        selected={assigned}
+        setSelected={(element) => {
+          if (assigned.some((other) => element.id === other.id)) {
+            setAssigned((prev) =>
+              prev.filter((other) => element.id !== other.id)
+            );
+          } else {
+            setAssigned((prev) => [...prev, element]);
+          }
+        }}
+        onSave={onSave}
+      />
+      {assigned.map((eachSenior) => (
```

Comment:
We want to use `CardGrid` component here. The current styling is this

<img width="1365" alt="image" src="https://github.com/JumboCode/TheLegacyProject2/assets/74647679/bc3c9621-7d08-4d15-aea1-972611a90c76">